### PR TITLE
Eng 16116 move export junit to community

### PR DESCRIPTION
**Pull request by wweiss-voltdb**

### CODE REVIEW

File: src/frontend/org/voltdb/utils/CatalogSchemaTools.java [hunk 1]
```diff
@@ -152,7 +152,8 @@ public static String toSchema(StringBuilder sb, Table catalog_tbl, String viewQu
                 continue;
             }
 
-            table_sb.append(add).append(spacer).append(catalog_col.getTypeName()).append(" ").append(col_type.toSQLString()).append(col_type.isVariableLength() &&
```
Comment (manjujames) [inline src/frontend/org/voltdb/utils/CatalogSchemaTools.java]: It'll be helpful if we avoid whitespace changes or separate them to a "cleanup" type commit, especially for these kinds of pull requests that involve a large number of files.

File: tests/frontend/org/voltdb/TestExportRejoinWithView.java [hunk 2]
```diff
@@ -158,7 +155,6 @@ public void testStreamViewWithRejoin() throws Exception {
         client.callProcedure("@AdHoc", insertSql.toString());
         }
         client.drain();
-        waitForStreamedAllocatedMemoryZero(client);
```
Comment (manjujames) [inline tests/frontend/org/voltdb/TestExportRejoinWithView.java]: This line doesn't need to replaced with the new call?

File: tests/frontend/org/voltdb/compiler/TestVoltCompiler.java [hunk 3]
```diff
@@ -475,6 +477,7 @@ public void testExportTableCase() throws IOException {
         project.addSchema(TestVoltCompiler.class.getResource("ExportTester-ddl.sql"));
```
Comment (manjujames) [inline tests/frontend/org/voltdb/compiler/TestVoltCompiler.java]: This test is disabled in community (see the check at the beginning of the test). Is that what we want?

File: tests/frontend/org/voltdb/export/TestExportAlterStreamEndToEnd.java [hunk 4]
```diff
@@ -68,7 +71,7 @@ public void setUp() throws Exception
         builder.setDeadHostTimeout(30);
         // Each stream needs an exporter configuration
         String streamName = "t";
-
+        streamNames = new ArrayList<>(Arrays.asList(streamName));
```
Comment (manjujames) [inline tests/frontend/org/voltdb/export/TestExportAlterStreamEndToEnd.java]: You can simply do a streamNames.add(streamName) here?

File: tests/frontend/org/voltdb/export/TestExportBase.java [hunk 5]
```diff
@@ -131,26 +135,18 @@ protected void quiesce(final Client client)
     /*
      * Test suite boilerplate
      */
-    public static final ProcedureInfo[] PROCEDURES = {
-        new ProcedureInfo(Insert.class, new ProcedurePartitionData ("NO_NULLS", "PKEY", "1"),
+    public static final ProcedureInfo[] NONULLS_PROCEDURES = {
+        new ProcedureInfo(ExportInsertNoNulls.class, new ProcedurePartitionData ("NO_NULLS", "PKEY", "1"),
                 new String[]{"proc"}),
-        new ProcedureInfo(InsertBase.class, null, new String[]{"proc"}),
```
Comment (manjujames) [inline tests/frontend/org/voltdb/export/TestExportBase.java]: Were these not used? Why were they removed?

**Comment by manjujames** [comment]
Are the files moved from pro repo simple moves without any changes? If there are any changes that I should look at, how do I do that?

**Comment by manjujames** [comment]
I don't think I see TestExportGroups here, which was deleted from pro.

File: tests/frontend/org/voltdb/export/TestPersistentExport.java [hunk 6]
```diff
@@ -50,10 +52,11 @@
             "CREATE TABLE T1 EXPORT TO TARGET FOO1 ON INSERT, DELETE (a integer not null, b integer not nulL);" +
             " PARTITION table T1 ON COLUMN a;" +
             "CREATE TABLE T3 EXPORT TO TARGET FOO3 ON UPDATE (a integer not null, b integer not nulL);";
+    private static List<String> streamNames = new ArrayList<>();
 
     @Before
     public void setUp() throws Exception {
-
+        streamNames.addAll(Arrays.asList("T1", "T2"));
```
Comment (manjujames) [inline tests/frontend/org/voltdb/export/TestPersistentExport.java]: The streams created are T1 and T3, but the ones added here are T1 and T2. Something is not right?

File: tests/testprocs/org/voltdb_testprocs/regressionsuites/exportprocs/ExportInsertAllowNulls.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+            String a_pool_s,
+            String a_pool_max_s,
+            byte[] b_inline,
+            byte[] b_pool,
+            BigDecimal a_decimal,
+            GeographyPointValue a_geography_point,
+            GeographyValue a_geography
+            )
+    {
+        assert(tablename.equals("S_ALLOW_NULLS"));
```
Comment (manjujames) [inline tests/testprocs/org/voltdb_testprocs/regressionsuites/exportprocs/ExportInsertAllowNulls.java]: How can this assert pass and the 'if' right below not be true? How can it get into any of the 'else' clauses?

File: tests/testprocs/org/voltdb_testprocs/regressionsuites/exportprocs/ExportInsertFromTableSelectMP.java [hunk 8]
```diff
@@ -21,36 +21,36 @@
...
 
-    public static final SQLStmt checkStateStmt = new SQLStmt(
-            "SELECT status FROM PAUSE_RESUME WHERE STATUS_ID = ?;");
-    public static final SQLStmt insertStmt = new SQLStmt(
-            "INSERT INTO PAUSE_RESUME (status, status_id) VALUES (?, ?);");
-    public static final SQLStmt updateStmt = new SQLStmt(
-            "UPDATE PAUSE_RESUME set status=? where status_id=1;");
+    // This statement currently fails to compile. So it is commented out for now.
+    //private final SQLStmt i_insert_select_repl = new SQLStmt
+    //("INSERT INTO S_ALLOW_NULLS_REPL SELECT * FROM NO_NULLS_REPL;");
```
Comment (manjujames) [inline tests/testprocs/org/voltdb_testprocs/regressionsuites/exportprocs/ExportInsertFromTableSelectMP.java]: Is this something that should be enabled in the future?

File: tests/testprocs/org/voltdb_testprocs/regressionsuites/exportprocs/ExportInsertFromTableSelectMP.java [hunk 9]
```diff
@@ -21,36 +21,36 @@
...
+        if (targetStream.equals("S_ALLOW_NULLS_REPL")) {
+            //voltQueueSQL(i_insert_select_repl);
         }
-        return flag;
+        else if (targetStream.equals("S_ALLOW_NULLS")) {
+            voltQueueSQL(i_insert_select_part);
+        }
+        else {
+            assert(false);
+            throw new RuntimeException("Don't call this.");
```
Comment (manjujames) [inline tests/testprocs/org/voltdb_testprocs/regressionsuites/exportprocs/ExportInsertFromTableSelectMP.java]: It won't get to the throw statement.

File: tests/testprocs/org/voltdb_testprocs/regressionsuites/exportprocs/ExportInsertFromTableSelectSP.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        if (targetStream.equals("S_ALLOW_NULLS_REPL")) {
+            voltQueueSQL(i_insert_select_repl);
+        }
+        else if (targetStream.equals("S_ALLOW_NULLS")) {
+            voltQueueSQL(i_insert_select_part);
+        }
+        else {
+            assert(false);
+            throw new RuntimeException("Don't call this.");
```
Comment (manjujames) [inline tests/testprocs/org/voltdb_testprocs/regressionsuites/exportprocs/ExportInsertFromTableSelectSP.java]: Unreachable stmt.

File: tests/testprocs/org/voltdb_testprocs/regressionsuites/sqltypesprocs/Insert.java [hunk 11]
```diff
@@ -80,18 +80,6 @@ else if (tablename.equals("ALLOW_NULLS")) {
                          a_pool_s, a_pool_max_s, b_inline, b_pool, a_decimal,
                          a_geography_point, a_geography);
         }
-        else if (tablename.equals("NO_NULLS_GRP")) {
```
Comment (manjujames) [inline tests/testprocs/org/voltdb_testprocs/regressionsuites/sqltypesprocs/Insert.java]: I noticed the GRP ones deleted from all tests. Why was that removed?

File: tests/frontend/org/voltdb/TestExportRejoinWithView.java (hunk 2 above)
Comment (wweiss-voltdb) [reply tests/frontend/org/voltdb/TestExportRejoinWithView.java]: waitForStreamAllocatedMemoryZero does not check consistency, just that there are no EE buffers. The inserts are synchronous calls so they don't affect correctness.

File: tests/frontend/org/voltdb/export/TestExportAlterStreamEndToEnd.java (hunk 4 above)
Comment (wweiss-voltdb) [reply tests/frontend/org/voltdb/export/TestExportAlterStreamEndToEnd.java]: Setup is called before each test so that would work if a clear() was also called to remove any old values from a previous test.

File: tests/frontend/org/voltdb/export/TestExportBase.java (hunk 5 above)
Comment (wweiss-voltdb) [reply tests/frontend/org/voltdb/export/TestExportBase.java]: There was a common insert base class that was used for all streams and had statements for all streams. This implies that all the different stream permutations need to be defined in the schema which violates the 2 stream limit. I change the common class to a specialized class.

File: tests/frontend/org/voltdb/export/TestPersistentExport.java (hunk 6 above)
Comment (wweiss-voltdb) [reply tests/frontend/org/voltdb/export/TestPersistentExport.java]: Yep. You are right. The test was failing under pro. Thanks.

File: tests/testprocs/org/voltdb_testprocs/regressionsuites/sqltypesprocs/Insert.java (hunk 11 above)
Comment (wweiss-voltdb) [reply tests/testprocs/org/voltdb_testprocs/regressionsuites/sqltypesprocs/Insert.java]: GRP tests were originally added to test export groups. This was deprecated a long time ago but the tests still have both stream names even though there is no difference between behaviour.

File: tests/frontend/org/voltdb/compiler/TestVoltCompiler.java (hunk 3 above)
Comment (wweiss-voltdb) [reply tests/frontend/org/voltdb/compiler/TestVoltCompiler.java]: This is one of the few tests that creates more then 2 streams on purpose. It could be divided into 3 separate tests but it is only testing cases insensitivity in stream names so I did not think it was worth creating 2 extra ddl files.

File: tests/testprocs/org/voltdb_testprocs/regressionsuites/exportprocs/ExportInsertAllowNulls.java (hunk 7 above)
Comment (wweiss-voltdb) [reply tests/testprocs/org/voltdb_testprocs/regressionsuites/exportprocs/ExportInsertAllowNulls.java]: I was using the assert to see if the other paths were even used. It would appear that they are not so I can remove the other branches and leave the assert.

File: tests/testprocs/org/voltdb_testprocs/regressionsuites/exportprocs/ExportInsertFromTableSelectMP.java (hunk 8 above)
Comment (wweiss-voltdb) [reply tests/testprocs/org/voltdb_testprocs/regressionsuites/exportprocs/ExportInsertFromTableSelectMP.java]: It should be enabled when the ticket is closed. I added the ticket number to the comment.

File: tests/testprocs/org/voltdb_testprocs/regressionsuites/exportprocs/ExportInsertFromTableSelectMP.java (hunk 9 above)
Comment (wweiss-voltdb) [reply tests/testprocs/org/voltdb_testprocs/regressionsuites/exportprocs/ExportInsertFromTableSelectMP.java]: This was inherited from InsertBase.java. Not sure why it is there but will remove the assert.
